app/src/main/java/com/example/fittracker/MainActivityPresenter.java

```diff
@@ -0,0 +1,34 @@
+package com.example.fittracker;
+
+public class MainActivityPresenter implements LogInContract.Presenter{
+    private LogInContract.View mView;
+    private LogInContract.Model mModel;
+    public MainActivityPresenter(LogInContract.View view, LogInContract.Model model){
+        mView=view;
+        mModel=model;
+    }
+
+    @Override
+    public void onLogInPressed() {
+        if (mModel.isValidLogIn(mView.getEmail(),mView.getPassword())){
+            onValidLogin();
+        }else{
+            onInvalidLogin();
+        }
+    }
+
+    @Override
+    public void onValidLogin() {
+        mView.onValidLogin();
```

Comment:
this is not necessary, you can call the mView.onValidLogin() from the onLogInPressed() method

Reply:
Done